### PR TITLE
[ALCA] [LLVM16]Fix set but unused variables warnings

### PR DESCRIPTION
**Pull request by smuzaffar**
CLANG IBs (which now use clang16) has a lot of warnings like [a]. This PR fixes few of these

[a]
```
warning: variable 'XXX' set but not used [-Wunused-but-set-variable]
```

### CODE REVIEW

**Comment by tvami** [comment]
@cmsbuild , please test for CMSSW_13_1_CLANG_X

**Comment by smuzaffar** [comment]
please test

**Comment by tvami** [comment]
@cmsbuild , please test for CMSSW_13_1_CLANG_X

**Comment by tvami** [comment]
@cmsbuild , please test for CMSSW_13_2_CLANG_X

**Comment by smuzaffar** [comment]
please test for CMSSW_13_2_CLANG_X

File: Calibration/HcalCalibAlgos/test/GammaJetAnalysis.cc [hunk 1]
```diff
@@ -1491,7 +1467,6 @@ void GammaJetAnalysis::analyze(const edm::Event& iEvent, const edm::EventSetup&
                 }                                                                 // Test if element is from HCAL
                 else if (elements[iEle].type() == reco::PFBlockElement::HFHAD) {  // Element is HF
                   ntypes++;
```
Comment (tvami) [inline Calibration/HcalCalibAlgos/test/GammaJetAnalysis.cc:1469]: so `ntypes` should be removed too, right?

**Comment by smuzaffar** [comment]
please test

**Comment by tvami** [comment]
@cmsbuild , please test for CMSSW_13_2_CLANG_X

**Comment by tvami** [comment]
+alca
 * CLANG seems happy now

**Comment by rappoccio** [comment]
+1

**Pull request merged**